packages/primitives/src/Switch/Switch.tsx

```diff
@@ -0,0 +1,112 @@
+import * as React from 'react'
+import styled from '@emotion/styled'
+import { Theme, css } from 'theme-ui'
+import { SerializedStyles } from '@emotion/serialize'
+import {
+  StyledSystemProps,
+  variant,
+  sx,
+  shouldForwardProp,
+  styledSystemProps,
+  Box,
+} from '../Box/Box'
+import { Label } from '../Label'
+import { Input } from '../Input'
+
+function SwitchIndicator({
+  checked,
+  disabled,
+}: {
+  checked?: boolean
+  disabled?: boolean
+}) {
+  return (
+    <Box
+      as="span"
+      border="solid"
+      borderWidth={1}
+      opacity={disabled ? 0.7 : 1}
+      backgroundColor={checked ? 'blue.1' : 'grey.5'}
+      borderColor={checked ? 'blue.4' : 'grey.5'} // darkest blue
```

Comment:
@alexanderchan these need to allow any theme. Right now we are forcing the developers to use our Blue color scale. This should most likely read from a variant.